src/org/zaproxy/zap/extension/ascanrules/TestPathTraversal.java

```diff
@@ -433,9 +435,19 @@ public void scan(HttpMessage msg, String param, String value) {
             // this is nice because it means we do not have to guess any file names, and would only require one
             // request to find the vulnerability 
             // but it would be foiled by simple input validation on "..", for instance.
-        } catch (IOException e) {
-            log.warn("Error scanning parameters for Path Traversal: " + e.getMessage(), e);
-        }
+		} catch (SocketTimeoutException ste) {
+			log.info("A timeout occurred while checking [" + msg.getRequestHeader().getMethod() + "] ["
+					+ msg.getRequestHeader().getURI() + "], parameter [" + param + "] for Path Traversal. "
+					+ "The currently configured timeout is: "
+					+ Integer.toString(Model.getSingleton().getOptionsParam().getConnectionParam().getTimeoutInSecs()));
+			if (log.isDebugEnabled()) {
+				log.debug("Caught " + ste.getClass().getName() + " " + ste.getMessage());
+			}
+		} catch (IOException e) {
+			log.info("An error occurred while checking [" + msg.getRequestHeader().getMethod() + "] ["
+					+ msg.getRequestHeader().getURI() + "], parameter [" + param + "] for Path Traversal."
+					+ "Caught " + e.getClass().getName() + " " + e.getMessage());

```

Comment:
I figured if it's caught here it might be something else that's been overlooked (as this SocketTimeoutException was) so logging as part of the info message is a good move. If you disagree I can move it to debug similar to L444...